jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -35,15 +37,21 @@
 import org.eclipse.jetty.client.util.BufferingResponseListener;
 import org.eclipse.jetty.http.HttpField;
 import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.QuotedCSV;
+import org.eclipse.jetty.util.StringUtil;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
 public abstract class AuthenticationProtocolHandler implements ProtocolHandler
 {
     public static final int DEFAULT_MAX_CONTENT_LENGTH = 16*1024;
     public static final Logger LOG = Log.getLogger(AuthenticationProtocolHandler.class);
-    private static final Pattern AUTHENTICATE_PATTERN = Pattern.compile("([^\\s]+)\\s+(.*,\\s*)?realm=\"([^\"]*)\"\\s*,?\\s*(.*)", Pattern.CASE_INSENSITIVE);
-
+    
+    private static final Pattern PARAM_PATTERN = Pattern.compile("([^=]+)=([^=]+)?");
+    private static final Pattern TYPE_PATTERN = Pattern.compile("([^\\s]+)(\\s+(.*))?");
+    private static final Pattern MULTIPLE_CHALLENGE_PATTERN = Pattern.compile("(.*),\\s*([^=\\s,]+(\\s+[^=\\s].*)?)\\s*");
+    private static final Pattern BASE64_PATTERN = Pattern.compile("([^\\s,=]+=*)\\s*");
```

Comment:
I'm not sure I understand this regexp (again, why the space at the end ?).
A quick googling shows that regexps for base64 are typically way more complicated.
Perhaps we can very lenient, but then the regexp may be way simpler ?